### DISCUSSION TITLE
Problems with SAGA middleware in the new version

### DISCUSSION THREAD

**Discussion by 0Ieg** [reduxjs/redux-toolkit#3982] [Q&A]
Problems with TS. 
In the "@reduxjs/toolkit" version: "^1.9.7" everything worked

![3](https://github.com/reduxjs/redux-toolkit/assets/111610358/d87c3bf2-9ccd-4de1-8201-862913fcb0e6)
![toolkit 2](https://github.com/reduxjs/redux-toolkit/assets/111610358/5a948669-2a1f-4a04-b456-c4bfa44d21f6)
![all](https://github.com/reduxjs/redux-toolkit/assets/111610358/236b24f6-b9b4-4558-bb00-900464373119)

**Error**
![error](https://github.com/reduxjs/redux-toolkit/assets/111610358/4b845874-4b06-4889-a69d-5b13447ac4de)

**Comment by 0Ieg** [comment]
After the rollback, everything worked.

npm i @reduxjs/toolkit@1.9.7
npm i react-redux@8.1.3

**Reply by phryneas** [reply]
It should be working: https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgYwgOwGbAOYFcoCmAyjNAXAL5wZQQhwDkAAoQCa4AeAVgM4D0pCABsA1sBgMAUKEiwUhAIYxiC7AoCywVqyEEA7gsLVa9Bm04BaHqoVTJqND3jnkBKHAC8cABQBKTwB8cACskpJ8fPIESuQwABbk1mpwIFo6+oYE9uhOcEkaaboGRl7IispENpraRZl+4ZEgELho8OJw6HDx5CRk2Y7wTmSeKOhYeIS9hN4IknBwLm4ANHBzKYUZhABcPtgEMAAiBBgKuEIw1enFBP4eQXuHx6fnl7XTvgB0DshK3vmvmxuS0kFF8kiAA

Did you update redux-saga to the latest version as well?

**Reply by 0Ieg** [reply]
I tried it like that, just add another middleware. Yes, and redux-saga, I was creating a new project

**Reply by 0Ieg** [reply]
Everything works with older versions. I am completely satisfied with the old versions. It's just that when installing modules, you will have to specify their versions.
![current](https://github.com/reduxjs/redux-toolkit/assets/111610358/ce666e1f-9646-4161-8ba2-4dd24e9bf51e)

**Reply by phryneas** [reply]
> I tried it like that, just add another middleware. Yes, and redux-saga, I was creating a new project

I mean it's working there with saga. Please give an example project that shows this not working.

**Reply by 0Ieg** [reply]
Here with the LoginAPI method

https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAQwB7AgZzgMyhEcBEyqa+A3AFDkCmSkscAxhAHZrwAyEA5sMwIIAFAJJwAvHAAUAEwQwEALgDeVEAmAAbeWyi8uAGjhgEaNAHdoUrTB3MuAXwCUogHyLycOFCowArlGYS7h6IKOgAdJBsEgDkCD4wABYA9OrcvNEGMnIOQR5hiVQBXmguxWFZCDnBcGEMsgwJElRQOFAuTKwQ6lRhqVxNLdBhxZCsPRUOVXA5dpSUoHTwioxeslQAyurADFRwdlg4ePgAAl5SPkgAVmhJMBBdANbAMGSUHWxwvM-ACOrrcjBduI3B5gGg+PEEvIsL80FRyLN3vA0FsdmIVlQ1pttlQJCC4MwECAqPJYpCMkEvjAfn8AVQ9EEzj4dlA0EpcnB2AB5ADiQgAcvIJGw1k5XByPCLAWEwRDEujrD54dU7AzqtyeVyAKoAFSFUqoYvx1QNMvBkPRmFhyuCsw8s0clBoi0YLA+coSACUqOcWeiUTjhj7mc1qLRoPAkXBFBqBQYNdqdftxAGdmEEAxqW65nMFhHoxi1nxM6hmHsDrgCKdg1cbndHs9Xnn6Ms6up1AY5A8qABRABuzQAngYwPFy9hK-gmUgALRoBBcBBJKiYTBUTMkChhl1Rzg8fhoQfMBh8ADC6IYq0BxazzAAPMpVBorDZ9IZjGYLC-dHZnBJ8LG-IAPp8OsACa-KnvgOTbvmmA+Met4AFRwB6ADq9QJM0EgOPIPKFM0sjQPig7AFQ6hSHAXa9gOUCDv+gEgeBkH4PGaTMGh0DdlAMyUPBiGlihe68JxUDcRIGa3vI3q+P4OqDmAVB3jAClUBAmCcuxfCHseZ7OLh+HMIRdxQPiUYZjsJhAXc3ZluIpHkZRbbqBIwn8MIBiSaWEQIIOqQIFIUzAJgEkMJZaDWRAtkOMacAORRhjxK5vICjhUwOjm8zgPmrYsJgwBcH4Gwmbs+wTkc1bnLWtz3OoTwvOQzaRleGwLggACywBSFI3SmAgXgGMs6xtZ13W9f1pUVng0TTnObXRI12Utog7bjocBCzfOi7Lqu64wCQOZRltHVdT1VB9V48jDYuo1nRdQKFoC10nWN50TThTrhvQR0lReeUFUV-zQLi+JMiy7LVHEiTyB63q+s0ap7IjICneNl04S4ADax23WjVAALoIjB-EliwKGevcMAYTADTYQZBFQERplBPFlG-C5mPoZhdP47xOOo29XjDAhEgUxAVPczxn0uipilwP8azyXL4gyX4zBK0pstqRpbDA2EXDeArgLOOQQA

**Reply by 0Ieg** [reply]
I saw that everything was working in playground, decided to return the new version, but the error did not disappear
![all](https://github.com/reduxjs/redux-toolkit/assets/111610358/b4d5e283-3179-40cc-a1a6-c839c2bbfc8a)

**Reply by phryneas** [reply]
Do you maybe have conflicting versions of `redux` installed? Try a `npm ls redux` or `yarn why redux`
